### DISCUSSION TITLE
Continous HTML output with multiple source files

### DISCUSSION THREAD

**Discussion by dmenne** [quarto-dev/quarto-cli#3267] [Q&A] [answered]
With `bookdown::html_document2` I can work on individual Rmd files by commenting the unused files, and get ONE long contiguous HTML file when in the end all are uncommented. 

```
rmd_files: [
  "index.Rmd"
    ,"Exclusion.Rmd"
#    ,"Demographics.Rmd"
#    ,"Genotype.Rmd"
]
```

By default, Quarto always creates multiple files and inserts paging code.

How can I get one contiguous file with Quarto?

**Comment by mcanouil** [comment, marked as answer]
Use the include shortcode, see <https://quarto.org/docs/authoring/includes.html>.
Comments are HTML comments.

```md
---
title: Title
format: html
---

{{< include index.Rmd >}}
{{< include Exclusion.Rmd >}}
<!-- {{< include Demographics.Rmd >}} -->
<!-- {{< include Genotype.Rmd >}} -->
```

**Reply by dmenne** [reply]
That's "the old style" before `bookdown`. Too bad, because I cannot easily switch between .yaml method for development and continuous version for delivery.

**Reply by mcanouil** [reply]
I might have a workaround, but so need to test it first.

Could you explain your use case? 
I do not really understand the issue you have by having to use only the Quarto document and not a confit file.

**Reply by dmenne** [reply]
Not really that important. I had spent some time to port a `bookdown` document which had a grandfather using the "include" style and thought that I had missed something.

Nevertheless, I think something like `paging: false` would be interesting. Maybe it should only be activated when the whole project is build, so development could be in chunks.

**Reply by mcanouil** [reply]
You can make a feature request[^1], where you explain what you'd like, why it's needed, and suggest possible ways to implement it (user side mostly).

[^1]: A GitHub discussion in the "Ideas" category preferably with a tag/prefix in the title, such as `[Feature Request]` to let everyone clearly identify the object of the discussion.

**Reply by dmenne** [reply]
#3288

**Reply by mcanouil** [reply]
Here is a workaround based on the `knitr` engine:

````
---
title: Title
format: html
qmd-files:
  - part1.qmd
  - part2.qmd
---

```{r}
#| output: asis
cat(
  sapply(rmarkdown::metadata[["qmd-files"]], knitr::knit_child, quiet = TRUE),
  sep = "\n"
)
```

````

**Reply by dmenne** [reply]
Interesting concept! I fear that could result in difficult to trace errors, so let's consider it a proof-of-workaround.

**Reply by mcanouil** [reply]
You can improve the code, for example add some messages in the `sapply` loop to let you know which file is being processed.
